### DISCUSSION TITLE
New release schedule

### DISCUSSION THREAD

**Discussion by Novum** [Novum/vkQuake#568] [General]
@temx Should we do a new update soon? Does anything need to be hardened first?

**Comment by temx** [comment]
Yeah, there's a bunch of new features already (installer, scale support, demo seek) apart from fixes.
Any chance you could take a look into the open PRs? They should be independent from one another but better test the combination that gets in.

**Comment by Novum** [comment]
Done. I want to fix the sound CSQC issues Spike pointed out too.

**Comment by temx** [comment]
Suggested notes for user visible changes (not the best way to document them, but still):
* Windows installer, detects existing Quake installations
* `vkquake.pak` is embedded into the executable, external file no longer used
* Configuration is now saved to `vkQuake.cfg` instead of `config.cfg`
* Demo seek support (arrow keys seek 10s, shift+arrows seek 30s, `seek` command bindable to function keys: `bind F1 "seek -3"`)
* Demo recording is no longer interrupted when loading savegames
* `fastload` command loads savegame without reloading level (`bind F6 "save quick"; bind F9 "fastload quick"`)
* Underwater sound filter (`snd_waterfx 0` to disable)
* `viewsize 130` for clean screenshots (`+`/`-` keys to quickly select)
* Built-in zooming support (`togglezoom`, `+zoom`/`-zoom`, `zoom_fov`, `zoom_speed`)
* Model `.scale` support for mods/maps
* Lightstyles no longer saturate at 2x brightness
* CPU/GPU optimizations for maps with very large geometry/lightmap footprints
* Fixed pitch-black entities in some Arcane Dimension levels
* `r_speeds` works again, `r_showbboxes` shows edict numbers
* Various other bug fixes

Markup:
```
* Windows installer, detects existing Quake installations
* `vkquake.pak` is embedded into the executable, external file no longer used
* Configuration is now saved to `vkQuake.cfg` instead of `config.cfg`
* Demo seek support (arrow keys seek 10s, shift+arrows seek 30s, `seek` command bindable to function keys: `bind F1 "seek -3"`)
* Demo recording is no longer interrupted when loading savegames
* `fastload` command loads savegame without reloading level (`bind F6 "save quick"; bind F9 "fastload quick"`)
* Underwater sound filter (`snd_waterfx 0` to disable)
* `viewsize 130` for clean screenshots (`+`/`-` keys to quickly select)
* Built-in zooming support (`togglezoom`, `+zoom`/`-zoom`, `zoom_fov`, `zoom_speed`)
* Model `.scale` support for mods/maps
* Lightstyles no longer saturate at 2x brightness
* CPU/GPU optimizations for maps with very large geometry/lightmap footprints
* Fixed pitch-black entities in some Arcane Dimension levels
* `r_speeds` works again, `r_showbboxes` shows edict numbers
* Various other bug fixes
```

**Comment by Novum** [comment]
I have one more change I wanted to see if I can get it in today

**Comment by Novum** [comment]
@temx Obviously this hasn't happened because of me being busy with various things. I tried to parallelize the FTE particle code, but that is a can of worms.

Let's give this another go. Do you foresee more bug fixes for the indirect rendering?

**Comment by temx** [comment]
I've run a script on ~600 levels that loads the level under validation, runs for 1 second, pauses, takes a screenshot of the starting position with r_indirect 0/1 and complains if they differ. So at least that works (this wouldn't have caught #580). I did test all the new paths at some point. A few of the other features are probably riskier, but optional and have seen more testing. So I think it's probably OK to release.

**Comment by Novum** [comment]
Done :partying_face:

**Comment by andrei-drexler** [comment]
Hey! Just a small request - for features coming from Ironwail, please state so in the release notes. Even if they're not exactly rocket science, they do add up to a better experience (as the feedback on Discord would also seem to indicate). Going forward, I'd also appreciate it if you waited until new features have been actually shipped before adopting the code - having two public vkQuake versions now with features I haven't yet released is kind of awkward. Thanks!

**Comment by Novum** [comment]
Andrei I do appreciate the code you do and I wasn't even aware that we imported some of it at the time of release. I'll add it to the release notes. I guess @temx added those. Which ones are they exactly?

As for the timing, it was purely determined by my family being out of town and me actually having time to ship it. This isn't a professional enterprise ;)

**Comment by temx** [comment]
These three are direct ports:

* Underwater sound filter (`snd_waterfx 0` to disable) [[src1](https://github.com/andrei-drexler/ironwail/commit/1c0a514e37a67f84ac42f2a9072ea66c41ab6dcb), [src2](https://github.com/andrei-drexler/ironwail/commit/992440352cacc4573d58d121342705e945a9cec5)]
* `viewsize 130` for clean screenshots (`+`/`-` keys to quickly select) [[src](https://github.com/andrei-drexler/ironwail/commit/6730fce0be2059a3e3e75f165dd61b3172b9998c)]
* Built-in zooming support (`togglezoom`, `+zoom`/`-zoom`, `zoom_fov`, `zoom_speed`) [[src](https://github.com/andrei-drexler/ironwail/commit/f7768633ad83dc44b8ab2b0528ff9be173d42ee5)]

(the 1st one was added [by request](https://github.com/Novum/vkQuake/issues/518), the other two at my discretion)

This was ported from QSS (Shpoike):
* Model `.scale` support for mods/maps

This was done by Andrei directly (and shipped in QS before):
* Fixed pitch-black entities in some Arcane Dimension levels

There are other things not in the notes, but all individual commits should be credited.

**Comment by Novum** [comment]
Done. @andrei-drexler, does this satisfy you?

**Comment by andrei-drexler** [comment]
Of course, it's maybe even a bit overkill, but I appreciate it, thank you!
